jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletApiRequest.java

```diff
@@ -761,37 +764,54 @@ public PushBuilder newPushBuilder()
             referrer += "?" + query;
         pushHeaders.put(HttpHeader.REFERER, referrer);
 
-        StringBuilder cookieBuilder = new StringBuilder();
-        Cookie[] cookies = getCookies();
-        if (cookies != null)
-        {
-            for (Cookie cookie : cookies)
-            {
-                if (!cookieBuilder.isEmpty())
-                    cookieBuilder.append("; ");
-                cookieBuilder.append(cookie.getName()).append("=").append(cookie.getValue());
-            }
-        }
+        Cookie[] existing = getCookies();
+        List<Object> cookies = new ArrayList<>();
+        if (existing != null && existing.length > 0)
+            cookies.addAll(Arrays.asList(existing));
+
         // Any Set-Cookie in the response should be present in the push.
         for (HttpField field : _servletContextRequest.getServletContextResponse().getHeaders())
         {
             HttpHeader header = field.getHeader();
             if (header == HttpHeader.SET_COOKIE || header == HttpHeader.SET_COOKIE2)
             {
-                HttpCookie httpCookie;
-                if (field instanceof HttpCookieUtils.SetCookieHttpField set)
-                    httpCookie = set.getHttpCookie();
-                else
-                    httpCookie = SET_COOKIE_PARSER.parse(field.getValue());
-                if (httpCookie == null || httpCookie.isExpired())
+                HttpCookie httpCookie = (field instanceof HttpCookieUtils.SetCookieHttpField set)
```

Comment:
I don't think we have a test for the PushBuilder with expired cookies?

Reply:
we do now!